### PR TITLE
Adding account-only signup flow

### PR DESCRIPTION
**Pull request by donlair**
#### Changes proposed in this Pull Request
* This PR adds a case for user-only account flow for Jetpack SSO.
* Additional details at pau2Xa-38j-p2.
* TODO: this PR doesn't yet address the final redirect.

#### Testing instructions
* Checkout this PR and start Calypso
* Open `https://equal-chimpanzee.jurassic.ninja/wp-login.php` in an Incognito window.
* Click "Log in with WordPress.com"
* Click "Create a new account" and confirm that you're taken to `/start/account/PARAMS`
* Grab that path, including params, and paste add it onto `calypso.localhost:3000`
* Confirm that you're taken through the user-only flow.
* At this point it should redirect you to the Reader, but it needs to redirect you to the provided redirect url.

Related to #

### CODE REVIEW

**Comment by donlair** [comment]
@taggon I've chased this down to the point of updating the login state tree to include the redirect URL, but I'm running into a deadens.

From what I can tell, after login success the user is redirected to the Sanitized version of redirect_to in the state tree ([code](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/login/actions/reboot-after-login.ts#L25)). The state should be set on  LOGIN_REQUEST_SUCCESS, but it doesn't seem to be in this case.

I tried inspecting the response from the login API, but my api debug skills failed.

**Comment by taggon** [comment]
@donlair As I already told you, we will take a look at this in the next pair programming session. 😄

**Comment by donlair** [comment]
@taggon I made it through the issues I was having before, but I think it's still not quite right. Timezones have been tough - do you mind providing some feedback async?

Couple things to highlight:
* Some of the `redirect` checks I added, felt a little hacky for some reason. Is there a better way to get the redirect param to pass through?
* This flow now handles and executes the redirect, but the redirect returns a nonce error. Are you familiar with that? I'm not sure where to look to chase that down further.

**Comment by kraftbj** [comment]
This worked well for me. I didn't get the nonce error suggested; however, I switched to the PR of calypso at the first Calypso URL (e.g. the initial `/log-in`) instead of waiting for the Create New Account step. I don't know if that makes any real difference.

The only hiccup in the flow is the requirement to verify the account before allowing sign-on, but I think that's a reasonable step that's required and outside the scope of this PR. (Opening the verification link in a separate tab did result in the original tab automatically updating once verified to continue the sign on).

tl;dr: From my testing, this gets my green light. Thank you so much for the quick work on this.

**Review by kraftbj — APPROVED** [review]
I did not review the specific code, but approved based on testing. The flow worked as expected, though I tested by switching to the calypso.live variant (not the localhost) immediately upon hitting `/log-in`.

File: client/lib/login/index.js [hunk 1]
```diff
@@ -121,5 +121,15 @@ export function getSignupUrl(
 		signupUrl = `${ signupUrl }/wpcc?${ oauth2Params.toString() }`;
 	}
 
+	if (
+		includes( get( currentQuery, 'redirect_to' ), 'action=jetpack-sso' ) &&
```
Comment (taggon) [inline client/lib/login/index.js]: You may want to reuse the `redirectTo` variable here. 😄

File: client/signup/steps/user/index.jsx [hunk 2]
```diff
@@ -393,6 +395,9 @@ export class UserStep extends Component {
 		) {
 			return this.props.initialContext.query.oauth2_redirect;
 		}
+		if ( this.props.initialContext.canonicalPath.includes( '/start/account' ) ) {
```
Comment (taggon) [inline client/signup/steps/user/index.jsx]: I'm wondering if there is any reason to use `includes` rather than`startsWith`. 🤔

Comment (taggon) [reply client/signup/steps/user/index.jsx]: The `initialContext` prop may not be provided in some environments. (e.g. unit testing) To make it safer, consider using optional chaining operators.
```js
this.props.initialContext?.canonicalPath?.includes( '/start/account' )
```

**Review by taggon — COMMENTED** [review]
It worked as expected! 🎉  I just left small suggestions for the safer approach.

> Some of the redirect checks I added, felt a little hacky for some reason. Is there a better way to get the redirect param to pass through?

Your approach looks good to me. Do you concern any specific point?

> This flow now handles and executes the redirect, but the redirect returns a nonce error. Are you familiar with that? I'm not sure where to look to chase that down further.

I don't see the nonce error too. My assumption is the key might be expired while you were testing.

**Comment by donlair** [comment]
Thanks @taggon! I've implemented your suggestions and this PR is ready for another look.

**Review by taggon — APPROVED** [review]
LGTM :)

**Comment by donlair** [comment]
Rebase

**Comment by mreishus** [comment]
Is it possible this is the cause of https://github.com/Automattic/wp-calypso/issues/55219 ?
